vm/src/api.rs

```diff
@@ -645,36 +656,14 @@ impl<'vm, T> Pushable<'vm> for Vec<T>
     }
 }
 
-pub struct Userdata<T>(pub T);
-
-impl<T: VmType> VmType for Userdata<T> {
-    type Type = T::Type;
-}
-impl<'vm, T: ::vm::Userdata> Pushable<'vm> for Userdata<T> {
-    fn push(self, vm: &'vm Thread, stack: &mut Stack) -> Status {
-        let data: Box<::vm::Userdata> = Box::new(self.0);
-        let userdata = vm.alloc(stack, Move(data));
-        stack.push(Value::Userdata(userdata));
-        Status::Ok
-    }
-}
-impl<'vm, T: Clone + ::vm::Userdata> Getable<'vm> for Userdata<T> {
-    fn from_value(_: &'vm Thread, value: Variants) -> Option<Userdata<T>> {
-        match value.as_ref() {
-            ValueRef::Userdata(data) => data.downcast_ref::<T>().map(|x| Userdata(x.clone())),
-            _ => None,
-        }
-    }
-}
```

Comment:
I left out this implementation as it requires a clone bound which could be unintentionally expensive.